mesonbuild/interpreter.py

```diff
@@ -2645,7 +2662,7 @@ def func_import(self, node, args, kwargs):
     @stringArgs
     @noKwargs
     def func_files(self, node, args, kwargs):
-        return [mesonlib.File.from_source_file(self.environment.source_dir, self.subdir, fname) for fname in args]
+        return [FileHolder(mesonlib.File.from_source_file(self.environment.source_dir, self.subdir, fname), self.environment.source_dir, self.environment.build_dir) for fname in args]
```

Comment:
Unfortunately it is not that easy, there are many places where we do `isinstance(f, mesonlib.File)`, and they won't work any more because the type is now `FileHolder`.

Reply:
So basically, in every API that takes file objects you now have to unholder() them first in interpreter.

Reply:
@xclaesse I am willing to put in the work to create `FileHolder` if people think it is an improvement in the interpreter code.

Reply:
`interpreterbase.py` has a `method_call` method that deals with method calls to non-interpreterobjects.